src/main/scala/strawman/collection/mutable/ImmutableBuilder.scala

```diff
@@ -0,0 +1,19 @@
+package strawman
+package collection
+package mutable
+
+import scala.Unit
+
+/**
+  * Reusable builder for immutable collections
+  */
+abstract class ImmutableBuilder[-Elem, Coll](empty: Coll)
+  extends ReusableBuilder[Elem, Coll] {
+
+  protected var elems: Coll = empty
+
+  def clear(): Unit = { elems = empty }
+
+  def result(): Coll = elems
+
+}
```

Comment:
I also re-introduced this class which is useful to define builders for immutable collections.